src/digits/ff31.rs

```diff
@@ -56,7 +56,7 @@ macro_rules! fp31 {
             ///as the conversion to Montgomery form + multiplication is as fast as normal multiplication + reduction.
             ///
             ///If you are doing more than 1 multiplication, it's clearly a win.
-            #[derive(PartialEq, Eq, Ord, Clone, Copy)]
+            #[derive(Debug, PartialEq, Eq, Ord, Clone, Copy)]
```

Comment:
IMO we should have a debug impl on all structs because it allows people to print intermediate results in algorithms. Is there a reason we wouldn't want it?

Reply:
Previously we'd omitted Debug for types that held secret values. Perhaps that isn't valid thinking for a number library though...

Reply:
At this level there is no idea of secret and Debug might even be something you want on secret values for debug purposes.